From: Sr-0w <Sr-0w@users.noreply.github.com>
Date: Tue, 4 Aug 2026 01:38:00 +0200
Subject: [PATCH] arm64: dts: qcom: hotdog: enable volume keys

The OnePlus SM8150 downstream device tree wires Volume Up to PM8150 GPIO6.
Configure that pin as an active-low gpio-key, matching the existing SM8150
Xiaomi mainline implementation.

Volume Down is connected to the PM8150 RESIN input. Enable the existing PON
key nodes and assign KEY_VOLUMEDOWN to RESIN.
---
 .../boot/dts/qcom/sm8150-oneplus-hotdog.dts   | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts b/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
--- a/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
@@ -10,7 +10,9 @@
 /dts-v1/;
 
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/pinctrl/qcom,pmic-gpio.h>
 #include "sm8150-oneplus-common.dtsi"
 
 / {
@@ -27,6 +29,21 @@
 	oplus,dtsi_no = <19801>;
 	oplus,pcb_range = <0 55>;
 
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&volume_up_state>;
+
+		volume-up {
+			label = "volume_up";
+			gpios = <&pm8150_gpios 6 GPIO_ACTIVE_LOW>;
+			linux,input-type = <EV_KEY>;
+			linux,code = <KEY_VOLUMEUP>;
+			wakeup-source;
+			debounce-interval = <15>;
+		};
+	};
+
 	memory@80000000 {
 		reg = <0 0x80000000 0 0x3bb00000>;
 	};
@@ -141,6 +158,17 @@
 	};
 };
 
+&pm8150_gpios {
+	volume_up_state: volume-up-state {
+		pins = "gpio6";
+		function = "normal";
+		input-enable;
+		bias-pull-up;
+		power-source = <1>;
+		qcom,drive-strength = <PMIC_GPIO_STRENGTH_NO>;
+	};
+};
+
 &tlmm {
 	panel_avdd_default: panel-avdd-default-state {
 		pins = "gpio130";
@@ -287,6 +315,15 @@
 	status = "okay";
 };
 
+&pon_pwrkey {
+	status = "okay";
+};
+
+&pon_resin {
+	linux,code = <KEY_VOLUMEDOWN>;
+	status = "okay";
+};
+
 &mdss_dsi0_phy {
 	vdds-supply = <&vreg_l5a_0p875>;
 	status = "okay";
-- 
2.50.1
